hook_scripts/quarto_python_formatter.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# numpydoc ignore=GL08
+import re
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+from typing import List, Match
+
+
+def format_python_code(
+    code: str, black_args: List[str]
+) -> str:  # numpydoc ignore=RT01
+    """Format Python code using Black with custom arguments."""
+    try:
+        cmd = ["black", "-"] + black_args
+        result = subprocess.run(
+            cmd,
+            input=code,
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        return result.stdout
+    except subprocess.CalledProcessError:
+        print(
+            "Error: Failed to format Python code with Black.", file=sys.stderr
+        )
+        return code
+
+
+def replace_code_block(
+    match: Match[str], black_args: List[str]
+) -> str:  # numpydoc ignore=RT01
+    """Replace code block with formatted version."""
+    return f"{match.group(1)}\n{format_python_code(match.group(2), black_args)}{match.group(3)}"
+
+
+def process_file(
+    filepath: Path, black_args: List[str]
+) -> None:  # numpydoc ignore=RT01
+    """Process the given file, formatting Python code blocks."""
+    python_code_block_pattern = r"(```\{python\})(.*?)(```)"
+    try:
+        content = filepath.read_text()
+        formatted_content = re.sub(
+            python_code_block_pattern,
+            lambda m: replace_code_block(m, black_args),
+            content,
+            flags=re.DOTALL,
+        )
+
+        with tempfile.NamedTemporaryFile(mode="w", delete=False) as temp_file:
+            temp_file.write(formatted_content)
+            temp_filepath = Path(temp_file.name)
+
+        temp_filepath.replace(filepath)
+    except IOError as e:
+        print(f"Error processing file {filepath}: {e}", file=sys.stderr)
+        sys.exit(1)
+
+
+if __name__ == "__main__":
+    if len(sys.argv) < 3:
+        print(
+            'Usage: python hook_scripts/quarto_python_formatter.py "BLACK_ARGS" <filename1.qmd> [filename2.qmd ...]'
+        )
+        sys.exit(1)
+
+    black_args = sys.argv[1].split()
```

Comment:
Long run we may want to move to explicit arg parsing so that the hook's `args` entry can make the expected style of typical pre-commit hook `args:` lists, but that doesn't need to happen now.

Reply:
I could not figure out how to have arbitrarily many `black_args` and `files`.

Reply:
could handle with `argparser`, but again beyond the scope of this PR.

Reply:
I tried that for longer than I should have but couldn't get it to work.